sdk/program/src/nonce/state/mod.rs

```diff
@@ -2,21 +2,125 @@
 
 mod current;
 pub use current::{Data, DurableNonce, State};
-use serde_derive::{Deserialize, Serialize};
+use {
+    crate::hash::Hash,
+    serde_derive::{Deserialize, Serialize},
+};
 
 #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
 pub enum Versions {
+    Legacy(Box<State>),
+    /// Current variants have durable nonce and blockhash domains separated.
     Current(Box<State>),
 }
 
 impl Versions {
-    pub fn new_current(state: State) -> Self {
-        Self::Current(Box::new(state))
+    pub fn new(state: State, separate_domains: bool) -> Self {
```

Comment:
Will revise this once the feature is activated and the gating code is removed;
For now using the same term would be easier to find all instances.

Reply:
I was thinking it was most useful while the gating code is in place so that it's clear what domain/version is used where. Is `separate_domains` the "same term" you are referring to? I personally think it's a bit confusing because it refers to the feature gate as well as the new nonce domain.